drivers/include/esp8266.h

```diff
@@ -0,0 +1,1144 @@
+    ESP8266_FirmwareUpdate_GotEdition = 0x03,  /*!< We have firmware edition to download */
+    ESP8266_FirmwareUpdate_StartUpdate = 0x04, /*!< Update has started */
+} ESP8266_FirmwareUpdate_t;
+
+/**
+ * \brief  Sleep mode enumeration
+ */
+typedef enum {
+    ESP8266_SleepMode_Disable = 0x00, /*!< Sleep mode disabled */
+    ESP8266_SleepMode_Light = 0x01,   /*!< Light sleep mode */
+    ESP8266_SleepMode_Modem = 0x02    /*!< Model sleep mode */
+} ESP8266_SleepMode_t;
+
+/**
+ * \brief  IPD network data structure
+ */
+typedef struct {
+    uint8_t InIPD;        /*!< Set to 1 when ESP is in IPD mode with data */
+    uint16_t InPtr;       /*!< Input pointer to save data to buffer */
+    uint16_t PtrTotal;    /*!< Total pointer to count all received data */
+    uint8_t ConnNumber;   /*!< Connection number where IPD is active */
+    uint8_t USART_Buffer; /*!< Set to 1 when data are read from USART buffer or 0 if from temporary buffer */
+} ESP8266_IPD_t;
+
+/**
+ * \brief  Connection type
+ */
+typedef enum {
+    ESP8266_ConnectionType_TCP = 0x00, /*!< Connection type is TCP */
+    ESP8266_ConnectionType_UDP = 0x01, /*!< Connection type is UDP */
+    ESP8266_ConnectionType_SSL = 0x02  /*!< Connection type is SSL */
+} ESP8266_ConnectionType_t;
+
+/**
+ * \brief  Connection structure
+ */
+typedef struct {
+    uint8_t Number;              /*!< Connection number */
+    uint16_t RemotePort;         /*!< Remote PORT number */
+    uint8_t RemoteIP[4];         /*!< IP address of device */
+    ESP8266_ConnectionType_t Type; /*!< Connection type. Parameter is valid only if connection is made as client */
+    uint32_t BytesReceived;      /*!< Number of bytes received in current +IPD data package. U
+                                        Use @arg DataSize to detect how many data bytes are in current package when callback function is called for received data */
+    uint32_t TotalBytesReceived; /*!< Number of bytes received in entire connection lifecycle */
+#if ESP8266_USE_SINGLE_CONNECTION_BUFFER == 1
+    char* Data;                  /*<! Use pointer to data array */
+#else
+    char Data[ESP8266_CONNECTION_BUFFER_SIZE + 1]; /*!< Data array for connection */
+#endif
+    uint16_t DataSize;           /*!< Number of bytes in current data package.
+                                        Becomes useful, when we have buffer size for data less than ESP8266 IPD statement has data for us.
+                                        In this case, big packet from ESP8266 is split into several packages and this argument represent package size */
+    uint32_t ContentLength;      /*!< Value of "Content-Length" header if it exists in +IPD data packet */
+    char* Name;                  /*!< Pointer to connection name, useful when using as client */
+    void* UserParameters;        /*!< User parameters pointer. Useful when user wants to pass custom data which can later be used in callbacks */
```

Comment:
Couldn't use this to set a pointer to your `sock_udp_t` struct to avoid the global variables in `ESP8266_Callback_ClientConnectionConnected`?